pkg/client/googlegroup_github/mapper_test.go

```diff
@@ -24,6 +24,129 @@ import (
 	"github.com/abcxyz/pkg/testutil"
 )
 
+func TestCreateBidirectionalGoogleGroupGitHubMapper(t *testing.T) {
+	t.Parallel()
+	defaultWritePath := "test.textproto"
+	cases := []struct {
+		name                          string
+		fileReadpath                  string
+		content                       string
+		wantGoogleGroupToGitHubMapper *GroupMapper
+		wantGitHubToGoogleGroupMapper *GroupMapper
+		wantErr                       string
+	}{
+		{
+			name: "success",
+			content: `
+mappings: [
+  {
+    google_group: {
+      group_id: "test_id_1"
+    }
+    git_hub_team: {
+      org_id: 1
+      team_id: 2
+    }
+  },
+  {
+    google_group: {
+      group_id: "test_id_1"
+    }
+    git_hub_team: {
+      org_id: 1
+      team_id: 3
+    }
+  },
+  {
+    google_group: {
+      group_id: "test_id_2"
+    }
+    git_hub_team: {
+      org_id: 1
+      team_id: 4
+    }
+  },
+  {
+    google_group: {
+      group_id: "test_id_3"
+    }
+    git_hub_team: {
+      org_id: 1
+      team_id: 4
+    }
+  }
+]
+`,
+			wantGoogleGroupToGitHubMapper: &GroupMapper{
+				mappings: map[string][]string{
+					"test_id_1": {"1:2", "1:3"},
+					"test_id_2": {"1:4"},
+					"test_id_3": {"1:4"},
+				},
+			},
+			wantGitHubToGoogleGroupMapper: &GroupMapper{
+				mappings: map[string][]string{
+					"1:2": {"test_id_1"},
+					"1:3": {"test_id_1"},
+					"1:4": {"test_id_2", "test_id_3"},
+				},
+			},
+		},
+		{
+			name:         "file_not_exist",
+			fileReadpath: "not_exist_path",
+			wantErr:      "failed to read mapping file",
+		},
+		{
+			name:    "invalid_format",
+			content: `not valid`,
+			wantErr: "failed to unmarshal mapping file",
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			td := t.TempDir()
```

Comment:
Per my comment above, this test would be significantly simpler if you had the function accept an `io.Reader` or separated the file-parsing from the structure creation.